wordpress-mobile/WordPress-Android
# WP Stories: assigning temporary Id in FrameSaveStart event listener

Fixes #13662 

The crash observed in the related issue is due to a StoryFrameItem not having an id associated with it by the time it is **finished** being saved.

This works well when a Story is started from scratch, but the problem shows up when editing an existing Story block and adding a new slide to it.

The `StorySaveListener` is a component of the Gutenberg Editor (and hence is not present when writing a Story from scratch), and it basically handles updates for saving progress from the `FrameSaveService` and the Gutenberg Editor bridge. As such, it must have been assigned an id beforehand (like, at the start of being saved so we can correctly match save progress update events to the right Story Frame item and containing Story block)

This situation went like this:
- in `onStoryFrameSaveStart()` we're doing a `val localMediaId = event.frameId.toString()` which, if `event.frameId` is null (and it was), then `toString()` just converted it to a `"null"` String, and this is what was then passed on to the StoryMediaSaveListener which connected to the Gutenberg bridge. Nothing went wrong then afterwards, there simply wasn't a match for an id such as `"null"`.
- same thing in progress updates in `onStoryFrameSaveProgress()`, so nothing crashed there (and the event info hit the bridge only to have Gutenberg find no matches for id `"null"`
- finally, when we reached the `onStoryFrameSaveCompleted()` method, the non-null requirement is enforced and hence we crashed.

We do still want to crash if that is the case, what we need to do is to actually assign a temporary id at the beginning of the frame's save process, the same way we do when we start an empty Story block from the block editor from scratch.


To test:
1. create a story from scratch and publish it
2. go to the published Posts list
3. open the Story post you just published
4. when the Story composer appears, tap on the + control to add a new slide (that's on the right-bottom corner of the screen)
5. pick an image, add some text on it
6. tap on the ✅ icon on the top-right of the screen to save your changes
7. observe the crash doesn't happen anymore, and the Story gets correctly updated on your site (that is, you can see the new slide is there on your story on your site)


PR submission checklist:
